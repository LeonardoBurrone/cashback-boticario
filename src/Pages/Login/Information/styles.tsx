import { Paper, TextField } from '@material-ui/core';
import styled from 'styled-components';

const CBContainer = styled(Paper)`
  && {
    padding: 20px;
    &.center {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
    }
  }
`;

const CBTextField = styled(TextField)`
  && {
    min-width: 100%;
  }
`;

const Title = styled.p`
  align-items: flex-start;
  display: flex;
  flex-flow: column;
`;

export { CBContainer, CBTextField, Title };
