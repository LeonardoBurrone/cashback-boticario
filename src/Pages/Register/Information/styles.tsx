import { TextField } from '@material-ui/core';
import styled from 'styled-components';

const CBTextField = styled(TextField)`
  && {
    min-width: 100%;
  }
`;

export { CBTextField };
