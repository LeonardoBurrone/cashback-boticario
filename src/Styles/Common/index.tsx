import { Container, Paper, TextField } from '@material-ui/core';
import styled from 'styled-components';

const CBContainer = styled(Container)`
  && {
    display: flex;
    flex-flow: column;
  }
`;

const CBContent = styled(Paper)`
  && {
    padding: 20px;
    &.center {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
    }
  }
`;

const CBTextField = styled(TextField)`
  && {
    min-width: 100%;
  }
`;

const CBToolbar = styled('div')(({ theme }) => ({
  ...theme.mixins.toolbar
}));

export { CBContainer, CBContent, CBTextField, CBToolbar };
