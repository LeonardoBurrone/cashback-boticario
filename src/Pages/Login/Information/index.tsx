import { Grid } from '@material-ui/core';
import React, { useState } from 'react';

import { CBTextField } from './styles';

import ToggleVisibility from '../../../Components/ToggleVisibility';

type Props = {
  changeValue: (type: 'USERNAME' | 'PASSWORD', value: string) => void;
  password: string;
  passwordError: string;
  username: string;
  usernameError: string;
};

const Information: React.FunctionComponent<Props> = (props: Props) => {
  const [showPassword, setShowPassword] = useState<boolean>(true);

  const changeValue = (type: 'USERNAME' | 'PASSWORD') => (event: React.ChangeEvent<HTMLInputElement>) => {
    props.changeValue(type, event.target.value);
  };

  return (
    <Grid container spacing={1}>
      <Grid item xs={12}>
        <CBTextField
          autoFocus
          value={props.username}
          onChange={changeValue('USERNAME')}
          label="Usuário"
          id="username"
          margin="normal"
          helperText={props.usernameError || 'Insira o usuário'}
          error={props.usernameError.length > 0}
        />
      </Grid>
      <Grid item xs={12}>
        <CBTextField
          label="Senha"
          id="password"
          onChange={changeValue('PASSWORD')}
          margin="normal"
          type={showPassword ? 'password' : 'text'}
          helperText={props.passwordError || 'Insira a senha'}
          error={props.passwordError.length > 0}
          InputProps={{
            endAdornment: <ToggleVisibility show={showPassword} toggle={() => setShowPassword(!showPassword)} />
          }}
        />
      </Grid>
    </Grid>
  );
};

export default Information;
