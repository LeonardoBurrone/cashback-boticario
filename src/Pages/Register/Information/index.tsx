import DateFnsUtils from '@date-io/date-fns';
import { Grid } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
import React from 'react';

import { CBKeyboardDatePicker } from './styles';

import { CBTextField } from '../../../Styles/Common';

type Props = {
  changeValue: (type: 'code' | 'date' | 'price', value: string) => void;
  code: string;
  codeError: string;
  date: string;
  dateError: string;
  price: string;
  priceError: string;
};

const Information: React.FunctionComponent<Props> = (props: Props) => {
  const [selectedDate, setSelectedDate] = React.useState<MaterialUiPickersDate>(null);

  const changeValue = (type: 'code' | 'price') => (event: React.ChangeEvent<HTMLInputElement>) => {
    props.changeValue(type, event.target.value);
  };

  const handleDateChange = (date: MaterialUiPickersDate) => {
    if (date) {
      props.changeValue('date', date.toISOString());
      setSelectedDate(date);
    }
  };

  return (
    <Grid container spacing={1}>
      <Grid item xs={12}>
        <CBTextField
          autoFocus
          value={props.code}
          onChange={changeValue('code')}
          label="Código"
          id="code"
          margin="normal"
          helperText={props.codeError || 'Insira o código da compra'}
          error={props.codeError.length > 0}
        />
      </Grid>
      <Grid item xs={12}>
        <CBTextField
          label="Valor"
          id="price"
          onChange={changeValue('price')}
          margin="normal"
          type={'number'}
          helperText={props.priceError || 'Insira o valor da compra'}
          error={props.priceError.length > 0}
        />
      </Grid>
      <Grid item xs={12}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <CBKeyboardDatePicker
            disableToolbar
            variant="inline"
            format="MM/dd/yyyy"
            margin="normal"
            id="date-picker-inline"
            label="Selecione a data da compra"
            value={selectedDate}
            onChange={handleDateChange}
            KeyboardButtonProps={{
              'aria-label': 'change date'
            }}
          />
        </MuiPickersUtilsProvider>
      </Grid>
    </Grid>
  );
};

export default Information;
