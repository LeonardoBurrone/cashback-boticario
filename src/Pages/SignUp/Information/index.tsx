import { Grid } from '@material-ui/core';
import React, { useState } from 'react';

import ToggleVisibility from '../../../Components/ToggleVisibility';
import { CBTextField } from '../../../Styles/Common';

export type Props = {
  changeValue: (type: 'document' | 'email' | 'name' | 'password', value: string) => void;
  document: string;
  documentError: string;
  email: string;
  emailError: string;
  name: string;
  nameError: string;
  password: string;
  passwordError: string;
};

export const Information: React.FunctionComponent<Props> = (props: Props) => {
  const [showPassword, setShowPassword] = useState<boolean>(true);

  const changeValue = (type: 'document' | 'email' | 'name' | 'password') => (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    props.changeValue(type, event.target.value);
  };

  return (
    <Grid container spacing={1}>
      <Grid item xs={12}>
        <CBTextField
          autoFocus
          value={props.name}
          onChange={changeValue('name')}
          label="Nome completo"
          id="name"
          margin="normal"
          helperText={props.nameError || 'Insira seu nome completo'}
          error={props.nameError.length > 0}
        />
      </Grid>
      <Grid item xs={12}>
        <CBTextField
          value={props.document}
          onChange={changeValue('document')}
          label="CPF"
          id="document"
          margin="normal"
          helperText={props.documentError || 'Insira seu CPF'}
          error={props.documentError.length > 0}
        />
      </Grid>
      <Grid item xs={12}>
        <CBTextField
          value={props.email}
          onChange={changeValue('email')}
          label="E-mail"
          id="email"
          margin="normal"
          helperText={props.emailError || 'Insira seu e-mail'}
          error={props.emailError.length > 0}
        />
      </Grid>
      <Grid item xs={12}>
        <CBTextField
          label="Senha"
          id="password"
          onChange={changeValue('password')}
          margin="normal"
          type={showPassword ? 'password' : 'text'}
          helperText={props.passwordError || 'Insira sua senha'}
          error={props.passwordError.length > 0}
          InputProps={{
            endAdornment: <ToggleVisibility show={showPassword} toggle={() => setShowPassword(!showPassword)} />
          }}
        />
      </Grid>
    </Grid>
  );
};

export default Information;
