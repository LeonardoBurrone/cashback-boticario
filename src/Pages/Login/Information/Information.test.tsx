import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Information, { Props } from './';

import { CBTextField } from '../../../Styles/Common';

describe('Test component Information', () => {
  const changeValueSpy = jest.fn(() => ({}));

  const props: Props = {
    changeValue: changeValueSpy,
    email: '',
    emailError: '',
    password: '',
    passwordError: ''
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Information {...props} />);
  });

  it('Check if component rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Fill out inputs', () => {
    const emailEvent = {
      preventDefault() {},
      target: { value: 'test@test.com' }
    };
    const passwordEvent = {
      preventDefault() {},
      target: { value: '123456' }
    };

    wrapper.find(CBTextField).at(0).simulate('change', emailEvent);
    wrapper.find(CBTextField).at(1).simulate('change', passwordEvent);

    expect(changeValueSpy).toBeCalledWith('email', 'test@test.com');
    expect(changeValueSpy).toBeCalledWith('password', '123456');
  });
});
