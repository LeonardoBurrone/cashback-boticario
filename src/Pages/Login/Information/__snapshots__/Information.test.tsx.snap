// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Information Check if component rendered correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Styled(WithStyles(ForwardRef(TextField)))
      autoFocus={true}
      error={false}
      helperText="Insira seu e-mail"
      id="email"
      label="E-mail"
      margin="normal"
      onChange={[Function]}
      value=""
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Styled(WithStyles(ForwardRef(TextField)))
      InputProps={
        Object {
          "endAdornment": <ToggleVisibility
            show={true}
            toggle={[Function]}
          />,
        }
      }
      error={false}
      helperText="Insira a senha"
      id="password"
      label="Senha"
      margin="normal"
      onChange={[Function]}
      type="password"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
